### PR TITLE
Fix build warnings

### PR DESCRIPTION
**Pull request by kishore-sn**
Fix build warnings during build of loadable_elf_apps and
nxp_demo_protected configurations.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [e2fa8cfaa6545c3adb8f1773130957aca3adf192] -

**Comment by seinfra** [comment]
Target : [e2fa8cfaa6545c3adb8f1773130957aca3adf192] -

File: os/arch/arm/src/imxrt/imxrt_start.c [hunk 1]
```diff
@@ -316,8 +316,8 @@ static void go_os_start(void *pv, unsigned int nbytes)
  *   Make configuration to use the entire 512KB FlexRAM as OCRAM
  *
  ****************************************************************************/
-
-void imxrt_configure_ocram()
+#ifndef CONFIG_ARMV7M_DTCM
```
Comment (sunghan-chang) [inline os/arch/arm/src/imxrt/imxrt_start.c]: Is it right? not CONFIG_IMXRT_DTCM?

Comment (kishore-sn) [reply os/arch/arm/src/imxrt/imxrt_start.c]: Yes.. Either dtcm or ocram is configured. 
However, I modified the code slightly as below:

`#ifdef DTCM'
'void imxrt_configure_dtcm()'
'#else'
'void imxrt_configure_ocram()'

**Comment by seinfra** [comment]
Target : [76e4c6c50ba1e2a6e07219c96d1731180cef01d6] -

**Comment by seinfra** [comment]
Target : [76e4c6c50ba1e2a6e07219c96d1731180cef01d6] -

**Pull request merged**